### PR TITLE
Add filter method for s2s keys

### PR DESCRIPTION
**Pull request by litvaOo**
resolves https://github.com/nutanix/nunet-ui/issues/412

### CODE REVIEW

**Comment by litvaOo** [comment]
@shlomi-nx we need to restart this build after CI is fixed

File: papiea-engine/src/provider/provider_routes.ts [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
     const providerApiRouter = express.Router();
 
+    const filterKeys = async function (user: UserAuthInfo, filter: any): Promise<any> {
+        const result: any[] = await providerApi.filter_keys(user, filter);
+
+        const uuidToEntity: { [key: string]: any } = {};
+
+        result.forEach(x => {uuidToEntity[x.uuid] = x});
+
+        const entities = Object.values(uuidToEntity);
```
Comment (adolgarev) [inline papiea-engine/src/provider/provider_routes.ts]: why this is needed? result -> entities? results should contain only unique s2s keys already.

Comment (litvaOo) [reply papiea-engine/src/provider/provider_routes.ts]: Fixed

File: papiea-engine/src/provider/provider_routes.ts [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import * as express from "express";
 import { Provider_API, Provider_Power } from "./provider_api_interface";
-import { asyncHandler } from '../auth/authn';
+import { asyncHandler, UserAuthInfo } from '../auth/authn';
```
Comment (adolgarev) [inline papiea-engine/src/provider/provider_routes.ts]: is this used?

File: papiea-engine/src/provider/provider_routes.ts [hunk 3]
```diff
@@ -1,8 +1,9 @@
...
 import { BadRequestError } from '../errors/bad_request_error';
 
+export type SortParams = { [key: string]: number };
```
Comment (adolgarev) [inline papiea-engine/src/provider/provider_routes.ts]: is this used?

File: papiea-engine/src/provider/provider_routes.ts (hunk 2 above)
Comment (litvaOo) [reply papiea-engine/src/provider/provider_routes.ts]: Fixed

File: papiea-engine/src/provider/provider_routes.ts (hunk 3 above)
Comment (litvaOo) [reply papiea-engine/src/provider/provider_routes.ts]: Fixed

**Pull request merged**